sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -753,6 +753,24 @@ object ScalaReflection extends ScalaReflection {
     }
   }
 
+  /**
+   * [SPARK-20384] Create an underlying param for a given parameter of value class.
+   * When a member of case class is value class `extends AnyVal`, the member's parameter type
+   * for encoder should be the underlying type. This is to be consistent with the generated
+   * type of that member, and avoid compile error.
+   * @param param param (type is consistent with [[ScalaReflection.getConstructorParameters]])
+   * @return unwrapped param
+   */
+  private def unwrapValueClassParam(param: (String, `Type`)): (String, `Type`) = {
+    val (name, typ) = param
```

Comment:
nit: maybe use `tpe` instead of `typ` to be consistent with the naming in this file